autocrit.py

```diff
@@ -0,0 +1,158 @@
+import os
+import random
+import time
+
+import numpy as np
+import openai
+import torch
+from accelerate import Accelerator
+from datasets import load_dataset
+from torch.utils.data import DataLoader
+from tqdm import tqdm, trange
+from transformers import AutoModelForCausalLM, AutoTokenizer, DataCollatorWithPadding
+from itertools import cycle, islice
+
+from rich.console import Console
+
+console = Console(width=80)
+print = console.print
+print0 = lambda *args, **kwargs: print(*args, **kwargs) if os.environ.get("RANK", "0") == "0" else None
+
+@torch.inference_mode()
+def generate(model, tokenizer, prompts, temperature=1, max_new_tokens=256, max_length=2048, stop=[]):
+    inputs = tokenizer(prompts, return_tensors="pt", padding=True, truncation=True, max_length=max_length).to(model.device)
+    all_ids = model.generate(**inputs, temperature=temperature, max_new_tokens=max_new_tokens, do_sample=True, use_cache=True,
+                             pad_token_id=tokenizer.pad_token_id, eos_token_id=tokenizer.eos_token_id)
+    output_ids = all_ids[:, inputs.input_ids.shape[1]:]
+    outputs = tokenizer.batch_decode(output_ids, skip_special_tokens=True)
+
+    for i in range(len(outputs)):
+        for s in stop:
+            if s in outputs[i]:
+                outputs[i] = outputs[i][:outputs[i].index(s)]
+
+    return outputs
+
+def generate_openai(prompt, model="gpt-3.5-turbo", max_new_tokens=128, system_prompt="", temperature=1, stop=[]):
```

Comment:
This should be refactored with the inference backend system (once that is stable)